modules/database/variables.tf

```
@@ -62,6 +62,42 @@ variable "long_term_retention_policy_week_of_year" {
   default     = 1
 }
 
+variable "diagnostic_setting_name" {
+  description = "the name of this diagnostic setting"
+  type        = string
+  default     = "audit-logs"
+}
+
+variable "log_analytics_workspace_id" {
+  description = "The ID of the Log Analytics workspace to send diagnostics to."
+  type        = string
+}
+
+variable "log_analytics_destination_type" {
+  description = "The type of log analytics destination to use for this Log Analytics Workspace."
+  type        = string
+  default     = null
+}
+
+variable "diagnostic_setting_enabled_log_categories" {
+  description = "A list of log categories to be enabled for this diagnostic setting."
+  type        = list(string)
+
+  default = [
+    "SQLInsights",
+    "AutomaticTuning",
+    "QueryStoreRuntimeStatistics",
+    "QueryStoreWaitStatistics",
+    "Errors",
+    "DatabaseWaitStatistics",
+    "Timeouts",
+    "Blocks",
+    "Deadlocks",
+    "DevOpsOperationsAudit",
+    "SQLSecurityAuditEvents"
+  ]
```

Set only the log categories required for audit logging:

```suggestion
  default = [
    "DevOpsOperationsAudit",
    "SQLSecurityAuditEvents"
  ]
```